Lib/inspect.py

```diff
@@ -1341,6 +1341,9 @@ def getcallargs(*func_and_positional, **named):
     A dict is returned, with keys the function argument names (including the
     names of the * and ** arguments, if any), and values the respective bound
     values from 'positional' and 'named'."""
+    warnings.warn("`getcallargs` is deprecated since Python 3.5. Use"
+                  " `Signature.bind` and `Signature.bind_partial` instead.",
+                  DeprecationWarning, stacklevel=2)
     func = func_and_positional[0]
     positional = func_and_positional[1:]
     spec = getfullargspec(func)
```

Comment:
By the time I commented in the issue I don't think there were any code level warnings. But `getfullargspec` used in `getcallargs` was recently changed to generate a `DeprecationWarning` from d5d2b4546939b98244708e5bb0cfccd55b99d244. Generating two warnings seems little confusing to me.

cc: @pablogsal 

```
./python.exe -Wall
Python 3.8.0a3+ (heads/master:678bb9d1b5, May  6 2019, 23:38:09)
[Clang 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import inspect
>>> def foo(a, b): pass
...
>>> inspect.getcallargs(foo, 1, 2)
/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/inspect.py:1346: DeprecationWarning: Use inspect.signature() instead of inspect.getfullargspec()
  spec = getfullargspec(func)
{'a': 1, 'b': 2}
```

With PR : 

```
./python.exe -Wall
Python 3.8.0a3+ (heads/master:678bb9d1b5, May  6 2019, 23:38:09)
[Clang 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import inspect
>>> def foo(a, b): pass
...
>>> inspect.getcallargs(foo, 1, 2)
<stdin>:1: DeprecationWarning: `getcallargs` is deprecated since Python 3.5. Use `Signature.bind` and `Signature.bind_partial` instead.
/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/inspect.py:1349: DeprecationWarning: Use inspect.signature() instead of inspect.getfullargspec()
  spec = getfullargspec(func)
{'a': 1, 'b': 2}
```

Reply:
I concur with @tirkarthi, having two warnings is undesirable and leaks the fact that one function is using another. We could just silence one of the warnings or make `getfullargspec` a thing wrapper over the current implementation (renaming it to `_ getfullargspec`) to avoid the duplicate warning.

Reply:
Silencing warnings seems like a better idea to me instead of having `getfullargspec` call a wrapper function `_getfullargspec` and using `_getfullargspec` in `getcallargs` . As noted in the issue there was a discussion at https://bugs.python.org/issue27172#msg282174 about some changes around functionality difference between `getcallargs` and `Signature.bind`. 

I would request @ncoghlan to have a review over code level deprecation to see if he has any concerns.